tests/hotplug/affinity.go

```diff
@@ -0,0 +1,229 @@
+package hotplug
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"kubevirt.io/kubevirt/tests/libvmi"
+
+	"kubevirt.io/kubevirt/tests/testsuite"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	. "kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libwait"
+)
+
+var _ = Describe("[sig-compute]VM Affinity", decorators.SigCompute, decorators.SigComputeMigrations, decorators.RequiresTwoSchedulableNodes, Serial, func() {
+	var (
+		virtClient kubecli.KubevirtClient
+	)
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+
+	})
+
+	Context("Updating VMs node affinity", func() {
+		It("should successfully update node selector", func() {
+
+			By("Creating a running VM")
+			vmi := libvmi.NewAlpineWithTestTooling(
+				libvmi.WithMasqueradeNetworking()...,
+			)
+			vmi.Namespace = testsuite.GetTestNamespace(vmi)
+			vmi.Spec.Domain.CPU = &v1.CPU{
+				Sockets: 1,
+				Cores:   2,
+				Threads: 1,
+			}
+			vm := tests.NewRandomVirtualMachine(vmi, true)
+			vm.Spec.LiveUpdateFeatures = &v1.LiveUpdateFeatures{
+				Affinity: &v1.LiveUpdateAffinity{},
+			}
+
+			vm, err := virtClient.VirtualMachine(vm.Namespace).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(beReady())
+			libwait.WaitForSuccessfulVMIStart(vmi)
```

Comment:
Stupid question but I assume we have to wait for both because the VMI may not exist at first?

Reply:
@lyarwood I don't think I understand... 
We're waiting for both VM and VMI. What are we missing here?

Reply:
Again apologies if I wasn't clear, I was asking if just calling `libwait.WaitForSuccessfulVMIStart(vmi)` was enough here.

Reply:
Yes, I'm sorry. I was thinking that you are hinting about a possible race somewhere. 
We can wait for the VMI as this wait function gets it by name. 
I'd prefer to wait for the VM readiness and not to fail on `WaitForSuccessfulVMIStart` even if it's probably not likely to ever happen.